Resolve deprecated flag warning

## Hello!
<!-- Please leave this section as-is, it's designed to help others in the community know how to interact with our GitHub issues. -->

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details
Add deprecated flag to functions with deprecated.
```
providers/pulumi-aws/sdk/java/src/main/java/io/pulumi/aws/secretsmanager/outputs/GetSecretResult.java:57: warning: [dep-ann] deprecated item is not annotated with @Deprecated
    private final String rotationLambdaArn;
                         ^
providers/pulumi-aws/sdk/java/src/main/java/io/pulumi/aws/secretsmanager/outputs/GetSecretResult.java:65: warning: [dep-ann] deprecated item is not annotated with @Deprecated
    private final List<GetSecretRotationRule> rotationRules;
                                              ^
providers/pulumi-aws/sdk/java/src/main/java/io/pulumi/aws/secretsmanager/outputs/GetSecretResult.java:141: warning: [dep-ann] deprecated item is not annotated with @Deprecated
    public Boolean getRotationEnabled() {
                   ^
```
<!-- Please provide a general summary of the issue, including what you're trying to accomplish. -->

### Steps to reproduce
<!-- Provide a link to a live example or an unambiguous set of steps to reproduce this bug. Tell us what you expected to happen, and what happened instead. Please include code to reproduce, if relevant. -->

1.
2.
3.
4.

Expected: <!-- What did you expect to happen? -->
Actual: <!-- What happens instead? -->

--- a/pkg/codegen/jvm/gen.go
+++ b/pkg/codegen/jvm/gen.go
@@ -501,9 +501,7 @@ func (pt *plainType) genInputProperty(ctx *classFileContext, prop *schema.Proper
 		_, _ = fmt.Fprintf(w, "%s */\n", indent)
 	}
 
-	if prop.DeprecationMessage != "" {
-		_, _ = fmt.Fprintf(w, "%s@Deprecated\n", indent)
-	}
+	printObsoleteAttribute(ctx, prop.DeprecationMessage, indent)
 	_, _ = fmt.Fprintf(w, "%s@%s(name=\"%s\"%s)\n", indent, ctx.ref(names.InputImport), wireName, attributeArgs)
 	_, _ = fmt.Fprintf(w, "%sprivate final %s %s;\n", indent, propertyType.ToCode(ctx.imports), propertyName)
 	_, _ = fmt.Fprintf(w, "\n")
@@ -802,9 +800,7 @@ func (pt *plainType) genOutputType(ctx *classFileContext) error {
 			}
 			_, _ = fmt.Fprintf(w, "%s     */\n", indent)
 		}
-		if prop.DeprecationMessage != "" {
-			_, _ = fmt.Fprintf(w, "%s    @Deprecated\n", indent)
-		}
+		printObsoleteAttribute(ctx, prop.DeprecationMessage, indent+"    ")
 		_, _ = fmt.Fprintf(w, "%s    private final %s %s;\n", indent, fieldType.ToCode(ctx.imports), fieldName)
 	}
 	if len(props) > 0 {
@@ -935,9 +931,7 @@ func (pt *plainType) genOutputType(ctx *classFileContext) error {
 			}
 		}
 
-		if prop.DeprecationMessage != "" {
-			_, _ = fmt.Fprintf(w, "%s    @Deprecated\n", indent)
-		}
+		printObsoleteAttribute(ctx, prop.DeprecationMessage, indent+"    ")
 		if err := getterTemplate.Execute(w, getterTemplateContext{
 			Indent:          strings.Repeat("    ", 1),
 			GetterType:      getterType.ToCode(ctx.imports),
@@ -1214,9 +1208,7 @@ func (mod *modContext) genResource(ctx *classFileContext, r *schema.Resource, ar
 		outputParameterType := propertyType.ToCodeWithOptions(ctx.imports, TypeShapeStringOptions{
 			CommentOutAnnotations: true,
 		})
-		if prop.DeprecationMessage != "" {
-			_, _ = fmt.Fprintf(w, "    @Deprecated\n")
-		}
+		printObsoleteAttribute(ctx, prop.DeprecationMessage, "    ")
 		fmt.Fprintf(w,
 			"    @%s(name=\"%s\", type=%s, parameters={%s})\n",
 			ctx.ref(names.OutputExport), wireName, outputExportType, outputExportParameters)
@@ -1448,9 +1440,7 @@ func (mod *modContext) genFunction(ctx *classFileContext, fun *schema.Function, 
 	}
 
 	// Emit the datasource method.
-	if fun.DeprecationMessage != "" {
-		_, _ = fmt.Fprintf(w, "    @Deprecated\n")
-	}
+	printObsoleteAttribute(ctx, fun.DeprecationMessage, "    ")
 	_, _ = fmt.Fprintf(w, "    public static %s<%s> invokeAsync(%s@%s %s options) {\n",
 		ctx.ref(names.CompletableFuture), typeParameter, argsParamDef, ctx.ref(names.Nullable), ctx.ref(names.InvokeOptions))
 	_, _ = fmt.Fprintf(w,

--- a/pkg/codegen/testing/test/testdata/output-funcs-tfbridge20/jvm/src/main/java/io/pulumi/mypkg/GetAmiIds.java
+++ b/pkg/codegen/testing/test/testdata/output-funcs-tfbridge20/jvm/src/main/java/io/pulumi/mypkg/GetAmiIds.java
@@ -27,7 +27,7 @@ public class GetAmiIds {
  * aws.getAmiIds has been deprecated in favor of aws.ec2.getAmiIds
  * 
  */
-    @Deprecated
+    @Deprecated /* aws.getAmiIds has been deprecated in favor of aws.ec2.getAmiIds */
     public static CompletableFuture<GetAmiIdsResult> invokeAsync(GetAmiIdsArgs args, @Nullable InvokeOptions options) {
         return Deployment.getInstance().invokeAsync("mypkg::getAmiIds", TypeShape.of(GetAmiIdsResult.class), args == null ? GetAmiIdsArgs.Empty : args, Utilities.withVersion(options));
     }

